### PR TITLE
feat: Make Hcal insert's readout a hexagonal grid

### PR DESCRIPTION
**Pull request by rymilton**
### Briefly, what does this PR introduce?
Change the forward Hcal insert's readout to a hexagonal grid with hexagons of side size 1.889 cm and staggering between the layers. Test data was generated and it ran smoothly through EICrecon.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #496 )
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [X] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No breaking changes, no changes needed.
### Does this PR change default behavior?
Only changes the default readout of the insert.

### CODE REVIEW

**Comment by rymilton** [comment]
Depends on the implementation of a hexagonal grid segmentation class in DD4hep, as mentioned in Issue #496.

**Comment by rymilton** [comment]
In addition to changing to hexagonal readout, I am increasing the number of W/Sc layers from 4 to 10. This is the up-to-date design that is being used.

**Comment by rymilton** [comment]
Number of tungsten layers being moved to a separate PR.

File: compact/hcal/forward_insert.xml [hunk 1]
```diff
@@ -139,11 +139,16 @@
...
+      i.e. hexagons in layers 1-4 (5-8, ...) are all offset transversely from each other
+    </documentation>
     <readout name="HcalEndcapPInsertHits">
       <segmentation
-        type="CartesianGridXY"
-        grid_size_x="3.*cm"
-        grid_size_y="3.*cm"
+        type="HexGrid"
+        side_length="1.889*cm"
+        stagger="2"
```
Comment (wdconinc) [inline compact/hcal/forward_insert.xml:151]: Does this need a unit? Or is this a type of staggering?

Comment (rymilton) [reply compact/hcal/forward_insert.xml:151]: It's a type of staggering. The `stagger="2"` reflects the staggering described in lines 144/145.

Comment (rymilton) [reply compact/hcal/forward_insert.xml:155]: Alright, added more description and some links to the segmentation code.

**Review by Chao1009 — APPROVED** [review]
Looks good to me.

**Pull request merged**